app/uk/gov/hmrc/tariffclassificationfrontend/models/Permission.scala

```diff
@@ -16,36 +16,232 @@
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrTeamMembersOnly(operator)
+  }
 
-  private val teamBasicPermissions: Set[Permission] = Set(CREATE_CASES, VIEW_MY_CASES, VIEW_QUEUE_CASES, SEARCH, ADVANCED_SEARCH, VIEW_CASES, ASSIGN_CASE, VIEW_CASE_ASSIGNEE,
-    RELEASE_CASE, SUPPRESS_CASE, REOPEN_CASE, CANCEL_CASE, ADD_NOTE, ADD_ATTACHMENT, APPEAL_CASE, EXTENDED_USE, EDIT_SAMPLE)
+  case object REFER_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object REOPEN_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrTeamMembersOnly(operator)
+  }
+
+  case object REJECT_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object SUSPEND_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object COMPLETE_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object REMOVE_ATTACHMENTS extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object CANCEL_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrTeamMembersOnly(operator)
+  }
+
+  case object ADD_NOTE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrTeamMembersOnly(operator)
+  }
+
+  case object ADD_ATTACHMENT extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrTeamMembersOnly(operator)
+  }
+
+  case object KEYWORDS extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object EDIT_LIABILITY extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object EDIT_RULING extends CasePermission {
```

Comment:
Just wondering if you were planning to bring in the more complex permissions as part of this change, to make them part of the overall permissions trait? For instance the ones currently held in the LiabilityController along the lines of case is open and operator has permission for edit, **or**, case is new and operator has permission for create

Reply:
Stay tuned Aaron...  :-)

Reply:
Yep i'm going to do that as a second PR to keep them small

Reply:
Working on that next

Reply:
Sorry, the screen didn't refresh, I saw this afterwards -




> PR I'm going to pull out the case status logic in a separate